src/main/java/io/github/thebusybiscuit/slimefun4/implementation/operations/MiningOperation.java

```diff
@@ -22,20 +28,24 @@ public class MiningOperation implements MachineOperation {
 
     private final ItemStack result;
 
+    private final GEOResource resource;
+    private final Block block;
     private final int totalTicks;
     private int currentTicks = 0;
 
-    public MiningOperation(@Nonnull ItemStack result, int totalTicks) {
-        Validate.notNull(result, "The result cannot be null");
-        Validate.isTrue(totalTicks >= 0, "The amount of total ticks must be a positive integer or zero, received: " + totalTicks);
+    public MiningOperation(@Nonnull GEOResource resource, Block block, int totalTicks) {
```

Comment:
this is a public api, this will break other addons using this

Reply:
Can I make a ``class GEOMiningOperation extends MiningOperation``, then make the geo miner use that?